compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -92,6 +91,52 @@ impl<'a, 'tcx> Visitor<'tcx> for InferBorrowKindVisitor<'a, 'tcx> {
 }
 
 impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
+    fn construct_capture_info_string(
+        &self,
+        place: &Place<'tcx>,
+        capture_info: &ty::CaptureInfo<'tcx>,
+    ) -> String {
+        let variable_name = match place.base {
+            PlaceBase::Upvar(upvar_id) => var_name(self.tcx, upvar_id.var_path.hir_id).to_string(),
+            _ => bug!("Capture_information should only contain upvars"),
+        };
+
+        let mut projections_str = String::new();
+        for (i, item) in place.projections.iter().enumerate() {
+            let proj = match item.kind {
+                ProjectionKind::Field(a, b) => format!("({:?}, {:?})", a, b),
+                ProjectionKind::Deref => String::from("Deref"),
+                ProjectionKind::Index => String::from("Index"),
+                ProjectionKind::Subslice => String::from("Subslice"),
+            };
+            if i != 0 {
+                projections_str.push_str(",");
+            }
+            projections_str.push_str(proj.as_str());
+        }
+
+        let capture_kind_str = match capture_info.capture_kind {
+            ty::UpvarCapture::ByValue(value) => format!("{:?}", value),
+            ty::UpvarCapture::ByRef(borrow) => format!("{:?}", borrow.kind),
+        };
+        format!("Capturing {}[{}] -> {}", variable_name, projections_str, capture_kind_str)
```

Comment:
I'd not add Capturing here but in the function below, this would be useful because then we can just do 
`"Capturing + construct_capture ..." and `Min Capture + construct_capture ....`